### ISSUE TITLE
False positive in e18e/prefer-spread-syntax

### ISSUE THREAD

**Issue by colinscz** [e18e/eslint-plugin#132]
**e18e/prefer-spread-syntax** assumes that concat() is used for an Array when it is actually called upon a string in my scenario.

```typescript
 const stringValue = 'This is a test string';
 const data = stringValue.concat('This is the second part');
```

is turned into

```typescript
  const stringValue = 'This is a test string';
  const data = [...stringValue, 'This is the second part'];
```

I already forked the repo and started a [draft PR](https://github.com/e18e/eslint-plugin/pull/133), but I haven't found the exact point in code where I need to adjust the TypeChecker. Happy to help/assist and make a contribution.

**Comment by 43081j** [comment]
this seems odd. its already a type-enhanced rule and checks that the target is array-like.

https://github.com/e18e/eslint-plugin/blob/b6891741178d5930ef7fc48a496f4ee3d980036c/src/rules/prefer-spread-syntax.ts#L63-L65

**Comment by colinscz** [comment]
@43081j that is the part that I also checked...so it seems I was not far off, but if I put a test string with concat() into the corresponding list of "valid" and only run it with that one, then it fails at some point in isArrayType. Went debugging through it and at some point it identifies it as Array and returns true. 🤔

**Comment by 43081j** [comment]
when typescript-eslint isn't used, `isArrayType` returns `undefined` which means this rule treats `*.concat` as if it was an array.

that is expected behaviour and is probably what you're seeing. type enhanced rules basically work better when TSESLint is available.

with typescript-eslint, it should correctly infer that it is not an array and skip it.

**Comment by colinscz** [comment]
Hmm we are using typescript-eslint in our project combined with eslint and other eslint-plugins and it's all within an Nx monorepo. I will check maybe I can provide a small reproduction project next week.

**Comment by colinscz** [comment]
I think this issue can be addressed with this PR #135 by @webpro 💪🏽

**Comment by 43081j** [comment]
@colinscz even before Lars' PR this should've been fine for you, given you're using TSESLint.

im going to close this but if you do run into it again, please reply and we can reopen
